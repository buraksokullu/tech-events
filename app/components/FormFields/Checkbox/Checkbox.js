import React from 'react';
import PropTypes from 'prop-types';

import s from './Checkbox.scss';

const CheckBox = ({ input, label, id, value, ...args }) => (
  <div className={`${s.checkboxFieldHolder}`} role="presentation">
    <input type="checkbox" id={id} {...input} {...args} checked={value} />
    {label && (
      <label htmlFor={id} className={s.checkboxLabel}>
        {label}
      </label>
    )}
  </div>
);

CheckBox.propTypes = {
  input: PropTypes.node,
  label: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  value: PropTypes.string
};

CheckBox.defaultProps = {
  input: null,
  value: null
};

export default CheckBox;
