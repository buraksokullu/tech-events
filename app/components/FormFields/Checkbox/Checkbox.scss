.checkboxFieldHolder {
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin: 0px 10px 20px 10px;

  &.disabled {
    cursor: not-allowed;

    * {
      cursor: not-allowed;
    }
  }
  .checkboxLabel {
    margin-left: 10px;
  }

  input[type='checkbox'] {
    appearance: none;
    background-color: $white;
    border: 2px solid $grey60;
    border-radius: 6.7px;
    cursor: pointer;
    display: inline-block;
    height: 20px;
    outline: none;
    position: relative;
    width: 20px;
    border-radius: 8px;
    height: 24px;
    width: 24px;
    &:checked:before {
      background-size: 16px 13px;
      bottom: 3px;
      height: 13px;
      left: 2px;
      position: absolute;
      right: 2px;
      top: 3px;
      width: 16px;
    }

    &:checked {
      background: $orange;
      border: 2px solid $orange;
      &:before {
        content: '';
        display: block;
        background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='15' height='12' viewBox='0 0 15 12'%3E%3Cpath fill='%23FFF' fill-rule='nonzero' d='M13.927 2.2l-6.845 9.289a1.011 1.011 0 0 1-1.43.188L.764 7.769a1 1 0 0 1 1.25-1.562L6.09 9.468l6.227-8.451a1 1 0 1 1 1.61 1.183z'/%3E%3C/svg%3E%0A");
        background-repeat: no-repeat;
      }
    }

    &:disabled {
      background: $grey20;
      border: 2px solid $grey20;
      cursor: not-allowed;
      pointer-events: none;
    }

    &.error {
      background-color: $fireRed5;
      border: 2px solid $fireRed40;
    }
  }
}
