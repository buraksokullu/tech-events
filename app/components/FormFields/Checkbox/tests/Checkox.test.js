import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../Checkbox';

describe('Checkbox', () => {
  let checkboxProps = null;

  beforeEach(() => {
    checkboxProps = {
      children: 'children',
      input: 'children',
      label: 'label',
      id: 'id',
      value: 'value'
    };
  });

  it('Should render correctly', () => {
    const component = shallow(<Checkbox {...checkboxProps} />);

    expect(component).toMatchSnapshot();
  });
});
