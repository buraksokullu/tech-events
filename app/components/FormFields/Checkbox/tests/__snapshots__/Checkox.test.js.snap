// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Should render correctly 1`] = `
<div
  className="undefined"
  role="presentation"
>
  <input
    0="c"
    1="h"
    2="i"
    3="l"
    4="d"
    5="r"
    6="e"
    7="n"
    checked="value"
    id="id"
    type="checkbox"
  >
    children
  </input>
  <label
    htmlFor="id"
  >
    label
  </label>
</div>
`;
